kolibri/core/device/api.py

```diff
@@ -369,3 +374,28 @@ def list(self, request):
 
     def retrieve(self, request, pk):
         return Response(get_mounted_drive_by_id(pk)._asdict())
+
+
+class RemoteFacilitiesViewset(views.APIView):
+    """
+    Api to retrieve facilities information from a remote device
+    :param str baseurl: url of the server, including port to connect
+    :return : json object containing the list of facilities of the device, with their id, name, learner_can_sign_up and learner_can_login_with_no_password info
+    """
+
+    permission_classes = (IsNotAnonymous,)
+
+    def get(self, request):
+        baseurl = request.query_params.get(
+            "baseurl", request.build_absolute_uri("/")[:-1]
+        )
+        path = reverse("kolibri:core:publicfacility-list")
```

Comment:
Will need to `lstrip("/")` to ensure it concatenates properly with the baseurl below, in the case where the baseurl has a prefix.

See here where I've just fixed this for the same endpoint in 0.15: https://github.com/learningequality/kolibri/pull/9533/files#diff-d4ed59d24014dff73b18af199c3aa7484e644edae6e7cb162125ddf6d8c8f349R129

Reply:
thanks, I had considered it when using `build_absolute_uri`, but not when a parameter is added. Anyway I've prefered to use `rstrip` to limit `baseurl`. Addinig `lstrip` to the path could cause problems if baseurl does not have the ending `"/"`

Reply:
`rstrip` on the baseurl gives the wrong result though.

If you do this:
```
urljoin("http://127.0.0.1:8080/kolibri", "/api/public/facility")
```
then you get this output:
```
'http://127.0.0.1:8080/api/public/facility'
```
Which is precisely what we are trying to circumvent.

Reply:
Ok, I never thought `base_url` would have anything else but the server and the port. I've just changed it according to your suggestion.